rescript/plugin_setup.py

```diff
@@ -828,6 +830,34 @@
         "used for slicing the original alignment."),
 )
 
+T = TypeMatch([AlignedSequence])
```

Comment:
Quick comment on `TypeMatch`. I am in the process of figuring this out myself. I think that the [code here](https://github.com/qiime2/q2-feature-table/blob/9ed6160adad45445ec054e5ce034a3b3ba25a9b4/q2_feature_table/plugin_setup.py#L379), is a great example of what we'd want. This is part of the `qiime feature-table filter-seqs` command. Apologies, if you've already tried this.

Reply:
I've looked through many examples but did not see that one - thanks! I think it would work if we could stick with the `DNAIterator` as input type, however I did not want to use it as I need to iterate over the sequences twice for it to work the way I described above. That's why I'm creating the iterator "manually" within  `_read_seqs`... :/ 

You gave me another idea though - will try tomorrow!

Reply:
You just headed me off at the pass! I was just going to ask why you were not using DNAIterator. :-)

Reply:
I think you can just simply just set `seek(0)` after running through the first time with DNAIterator.

Reply:
OK, I think I got it to work (thanks for the hints @nbokulich and @mikerobeson!) - you be the judge if that makes sense. 

I don't think using `seek` works here - at least it didn't when I tried - am I missing something? Instead, I used the `tee` functionality from itertools to make a clone of the original iterator and that worked too. I ended up using `DNAIterator` as both, input and output, and created a transformer `DNAIterator` -> `AlignedDNAFASTAFormat` to allow converting back to appropriate output type. Let me know what you think. Thanks!

Tested with both input types on a file with 1mln 400bp-long sequences - worked as expected.

Reply:
This looks awesome @misialq! I'll take it for a spin now.